python/cuspatial/cuspatial/geometry/geoseries.py

```diff
@@ -0,0 +1,457 @@
+            + "\n"
+            + self.polygons.__repr__()
+            + "\n"
+        )
+
+
+class GpuPoints:
+    def __init__(self):
+        self.xy = cudf.Series([])
+        self.z = None
+        self.has_z = False
+
+    def __iter__(self):
+        self._index = 0
+        return self
+
+    def __next__(self):
+        if self._index >= len(self.xy):
+            raise StopIteration
+        result = self[self._index]
+        self._index = self._index + 2
+        return result
+
+    def __getitem__(self, index):
+        if isinstance(index, slice):
+            new_index = slice(index.start * 2, index.stop * 2 + 1, index.step)
+            return self.xy.iloc[new_index]
+        return self.xy.iloc[(index * 2) : (index * 2) + 2]
+
+    def __repr__(self):
+        return "xy: " + self.xy.__repr__()
+
+
+class GpuOffset(GpuPoints):
+    def __init__(self):
+        self.offsets = None
+
+    def __iter__(self):
+        if self.offsets is None:
+            raise TypeError
+        self._index = 1
+        return self
+
+    def __next__(self):
+        if self._index >= len(self.offsets):
+            raise StopIteration
+        result = self[self._index]
+        self._index = self._index + 1
+        return result
+
+    def __getitem__(self, index):
+        if isinstance(index, slice):
+            rindex = index
+        else:
+            rindex = slice(index, index + 1, 1)
+        new_slice = slice(self.offsets[rindex.start], None)
+        if rindex.stop < len(self.offsets):
+            new_slice = slice(new_slice.start, self.offsets[rindex.stop])
+        result = self.xy[new_slice]
+        return result
+
+    def __repr__(self):
+        return (
+            super().__repr__() + "\noffsets: " + self.offsets.__repr__() + "\n"
+        )
+
+
+class GpuLines(GpuOffset):
+    def __init__(self):
+        self.mlines = None
+
+    def __getitem__(self, index):
+        result = super().__getitem__(index)
+        return result
+
+    def __repr__(self):
+        return (
+            super().__repr__() + "\nmlines: " + self.mlines.__repr__() + "\n"
+        )
+
+
+class GpuMultiPoints(GpuOffset):
+    def __init__(self):
+        pass
+
+
+class GpuPolygons(GpuOffset):
+    def __init__(self):
+        self.polys = None
+        self.rings = None
+        self.mpolys = None
+
+    def __repr__(self):
+        result = ""
+        result += "xy:\n" + self.xy.__repr__() + "\n"
+        result += "polys:\n" + self.polys.__repr__() + "\n"
+        result += "rings:\n" + self.rings.__repr__() + "\n"
+        result += "mpolys:\n" + self.mpolys.__repr__() + "\n"
+        return result
+
+
+class cuGeometry:
+    def __init__(self, source, index):
+        self.source = source
+        self.index = index
+
+
+class cuPoint(cuGeometry):
+    def to_shapely(self):
+        item_type = self.source.types[self.index]
+        index = (
+            pd.Series(self.source.types[0 : self.index]) == item_type
+        ).sum()
```

Comment:
nit: Would probably be more efficient to avoid the pandas Series here and calculate the sum via normal iteration.